### PR TITLE
netlistsvg: init at 1.0.2

### PR DESCRIPTION
**Pull request by pbsds**
## Description of changes

[netlistsvg](https://github.com/nturley/netlistsvg) generates SVG diagrams of electrical circuits processed by yosys.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pbsds** [comment]
@ofborg build netlistsvg

**Review by D4ndellion — CHANGES_REQUESTED** [review]

**Comment by happysalada** [comment]
Could you open an issue upstream to ask them to commit their package-lock ?
I'm always a little sad when we have to add 9k lines to nixpkgs.

File: pkgs/applications/science/logic/netlistsvg/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  } ''
+    yosys -p "prep -top helloworld -flatten; aigmap; write_json circuit.json" ${./test.v}
+    netlistsvg circuit.json -o circuit.svg
+    test -s circuit.svg
+    touch $out
+  '';
+
+  meta = {
+    description = "Draw SVG digital circuits schematics from yosys JSON netlists";
+    homepage = "https://neilturley.dev/netlistsvg/";
```
Comment (dotlambda) [inline pkgs/applications/science/logic/netlistsvg/default.nix:50]: This is not a homepage, just a demo server.

Comment (pbsds) [reply pkgs/applications/science/logic/netlistsvg/default.nix:50]: Sorry about that. https://github.com/NixOS/nixpkgs/pull/254292

**Comment by dotlambda** [comment]
> Could you open an issue upstream to ask them to commit their package-lock ? I'm always a little sad when we have to add 9k lines to nixpkgs.

https://github.com/nturley/netlistsvg/issues/136

**Comment by dotlambda** [comment]
We should have waited with the merge until upstream has added the lock file.

**Comment by happysalada** [comment]
I thought about it, but the repo hasnt been active for a while, so i thought maybe people wouod genuinely benefit from this package before the lockfile gets added (it might take 6 months).

Would you want to remove this until they add it ?

**Comment by dotlambda** [comment]
> Would you want to remove this until they add it ?

No but we cannot keep adding vendored code whenever necessary. I just want people to say no in the future.

**Comment by happysalada** [comment]
Fine by me, i was on the fence. Im also aware of all the efforts you made toward removing those dependencies, thank you for all the work !

**Comment by dotlambda** [comment]
@pbsds How about you make an upstream PR adding a lock file so we can pull from that?

**Comment by pbsds** [comment]
Huh, I'm able to make a lock for 1.0.2, but not for the master branch. Neither with node 18 nor 20. Any older node in nixpkgs is marked vulnerable.

Their CI, which last ran four months ago, used node 14 and managed to install the deps.

**Pull request merged**